pkg/metrics/metrics.go

```diff
@@ -252,3 +297,28 @@ func RegisterAndExport(port string, log log.Logger) error {
 	log.Infof("Registered and exported metrics on port %s", port)
 	return nil
 }
+
+// ReportIMDSOperationError reports IMDS error count
+func (r *Reporter) ReportIMDSOperationError(operation string) error {
```

Comment:
Quick question - why is this preferable vs two methods - one which counts and one which measure latency ? and then separate flags which can be used in a switch case to perform different operations like - ImdsOperationsErrorCount